subsys/bluetooth/host/hci_core.c

```diff
@@ -749,6 +753,28 @@ static int hci_le_read_remote_features(struct bt_conn *conn)
 	return 0;
 }
 
+int hci_read_remote_version(struct bt_conn *conn, bt_remote_version_cb *rv_cb)
+{
+	struct bt_hci_cp_read_remote_version_info *cp;
+	struct net_buf *buf;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_READ_REMOTE_VERSION_INFO, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(conn->handle);
+	remote_version_cb = rv_cb;
+	int err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_REMOTE_VERSION_INFO, buf,
+				NULL);
+	if  (err != 0) {
+		return err;
```

Comment:
If send fails you would left remote_version_cb set here so Id leave it to be set later, though there is another problem where another thread may call hci_read_remote_version in the meantime and override the cb so we may need a semaphore either way to ensure only one thread is pending waiting the callback at the time.

Reply:
So you want this?

```
if  (err != 0) {
    remote_version_cb = NULL;
    return err;
}
```

Reply:
@Vudentz Considering what the function does, how long it takes (or rather doesn't) and how few people are actually going to use it, is that really necessary?

Reply:
@Vudentz What would be a suitable timeout for k_sem_take waiting for get_remote version to clear? 1-5 seconds maybe? (I'm thinking K_FOREVER is a probably a bad idea)

Reply:
I guess z_ is for use on userspace code, though I may be wrong here since I never really use them. Back the the timeout, we could use the same interval as bt_hci_cmd_send_sync uses.